### ISSUE TITLE
[MAUI] The list does not show newly added recipes

### ISSUE THREAD

**Issue by kevinxufei** [dotnet/maui#22417]
### Description

The list does not show newly added recipes

**Note:** The issue only repro on windows platform

### Steps to Reproduce

1. Clone https://github.com/rachelkang/recipeSearch and deploy it to windows platform
2. Navigate to My Recipes-> 'New', and add new recipe.
3. The list does not show newly added recipes

**Expected:** The list should show newly added recipes

**Actual:** The list does not show newly added recipes
![maui4](https://github.com/dotnet/maui/assets/86938401/3637f8a0-6dbc-496d-9870-f9448c6b14c6)

### Link to public reproduction project repository

https://github.com/rachelkang/recipeSearch.git

### Version with bug

Microsoft.Maui.Controls 8.0.60-ci.net8.10665+azdo.9579251

### Is this a regression from previous behavior?

This bug is a regression, it can be reproduced in 8.0.40, but can not be reproduced in 8.0.3, and it works fine on Microsoft.Maui.Controls 8.0.40-nightly.10612+sha.752eed7129-azdo.9538080.

### Affected platforms

Windows

**Comment by jsuarezruiz** [comment]
I have been reviewing it:
- We recently applied changes to the `CarouselViewHandler` https://github.com/dotnet/maui/pull/22222 We updated the ItemsSource under different conditions: changes in the size of the control or window, etc.
- When adding items, we no longer set the ItemsSource again and the UI does not refresh although internally everything is done correctly.
Using a `CollectionViewSource `with the ListViewBase of the CarouselView on Windows, everything works correctly although this would break the Loop property. Using `LoopableCollectionView`, the UI is not updated. Therefore, the focus is currently on reviewing this implementation. We could override and reset ItemsSource but it is a non-optimal solution.
